middleware/src/middleware.go

```diff
@@ -251,6 +256,37 @@ func (middleware *Middleware) Restore(method rpcmessages.RestoreArgs) (rpcmessag
 	}
 }
 
+// Hostname returns a GenericResponse struct in response to a rpcserver request
+func (middleware *Middleware) Hostname(args rpcmessages.HostnameArgs) (rpcmessages.GenericResponse, error) {
+	switch args.Method {
+	case rpcmessages.GetHostname:
+		log.Println("Getting the hostname via the config script")
+		out, err := middleware.runBBBConfigScript("get", "hostname", "")
+		if err != nil {
+			return rpcmessages.GenericResponse{Success: false, Message: string(out)}, err
+		}
+		return rpcmessages.GenericResponse{Success: true, Message: string(out)}, nil
+
+	case rpcmessages.SetHostname:
+		log.Println("Setting the hostname via the config script")
+		var r = regexp.MustCompile(`^[a-z0-9]+[a-z0-9-.]{0,62}[a-z0-9]$`)
+
+		if len(args.Hostname) >= 2 && len(args.Hostname) <= 64 && r.MatchString(args.Hostname) {
+			out, err := middleware.runBBBConfigScript("set", "hostname", args.Hostname)
+			if err != nil {
+				return rpcmessages.GenericResponse{Success: false, Message: string(out)}, err
+			}
+			return rpcmessages.GenericResponse{Success: true, Message: string(out)}, nil
+		}
+		errorMessage := fmt.Sprintf("invalid hostname '%s'", args.Hostname)
+		return rpcmessages.GenericResponse{Success: false, Message: errorMessage}, errors.New(errorMessage)
```

Comment:
fyi: if you return an error here, the client will get this as an error as well, so you don't need to replicate the error message in the response.

Important: `if err != nil`, then all other return values of a function (or rpc method) should be treated as undefined and not be touched. This makes the caller's life a lot easier.